src/Raven.Server/Commercial/LetsEncryptUtils.cs

```diff
@@ -0,0 +1,968 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Net.Http;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Org.BouncyCastle.Asn1;
+using Org.BouncyCastle.Asn1.X509;
+using Org.BouncyCastle.Crypto.Prng;
+using Org.BouncyCastle.OpenSsl;
+using Org.BouncyCastle.Pkcs;
+using Org.BouncyCastle.Security;
+using Raven.Client.Documents.Operations;
+using Raven.Client.ServerWide.Operations.Certificates;
+using Raven.Client.Exceptions;
+using Raven.Client.Exceptions.Security;
+using Raven.Server.Config;
+using Raven.Server.Config.Categories;
+using Raven.Server.ServerWide;
+using Raven.Server.Utils;
+using Raven.Server.Utils.Cli;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+using Sparrow.Platform;
+using Sparrow.Server.Platform.Posix;
+using Sparrow.Threading;
+using Sparrow.Utils;
+using StudioConfiguration = Raven.Client.Documents.Operations.Configuration.StudioConfiguration;
+
+namespace Raven.Server.Commercial
+{
+    public static class LetsEncryptUtils
```

Comment:
1. Why is the reason we had to move so many methods here? As I see most of them were moved from `SetupManager.cs` (which was already linked to `rvn` project BTW). Why they couldn't stay there?

2. Is `LetsEncryptUtils` a correct name for that class? I don't think it's specific to Let's Encrypt setup option? I thought it's more about the setup in general

3. We already have `CertificateUtils` class - I think that methods that deals specifically with certificates could be there.

Reply:
Making SetupManager file thinner is probably not a bad idea (it was over 2K liner itself), but if we're taking it out of there I would push it further and put classes in files of their own and also work on the proper naming then as Arek mentioned. Some of the methods here work with certs in general (if we need it separate from regular CertificateUtil maybe we need `LetsEncryptCertificateUtil`), some of them talk with LE API (all the CompleteChallenge*, Complete* and friends could go into `LetsEncryptApiHelper`) and some are DNS record change API (`RavenDnsRecordService/Helper`) consumers, another one is just concatenating a bunch of strings into a README (`SetupReadmeFileFactory`). Maybe instead of making it `LetsEncryptUtils` class we'd want a LetsEncrypt directory and put all those related classes there. If we're moving that in the first place, that's a great opportunity to make things better.